src/js/components/Select/Select.js

```diff
@@ -101,12 +101,18 @@ const Select = forwardRef(
         );
       return valueKey && valueKey.reduce ? value : applyKey(value, valueKey);
     }, [value, valueKey]);
-
+    // search input value
+    const [search, setSearch] = useState();
+    // track initialOptions for reference when option values are selected
+    // from filtered options set (e.g. such as when searching)
     const [initialOptions] = useState(options);
```

Comment:
I think initialOptions will never change. If the caller calls with a different `options`, `initialOptions` remains as it was initially.